api/users.go

```diff
@@ -53,35 +60,383 @@ func (self *ApiServer) SetPassword(
 	return &emptypb.Empty{}, users_manager.SetUser(ctx, user_record)
 }
 
+type userByName []*api_proto.VelociraptorUser
+
+func (s userByName) Len() int {
+    return len(s)
+}
+func (s userByName) Swap(i, j int) {
+    s[i], s[j] = s[j], s[i]
+}
+func (s userByName) Less(i, j int) bool {
+    return s[i].Name < s[j].Name
+}
+
 func (self *ApiServer) GetUsers(
 	ctx context.Context,
-	in *emptypb.Empty) (*api_proto.Users, error) {
+	in *api_proto.GetUsersRequest) (*api_proto.Users, error) {
 
 	users_manager := services.GetUserManager()
 	user_record, org_config_obj, err := users_manager.GetUserFromContext(ctx)
 	if err != nil {
 		return nil, Status(self.verbose, err)
 	}
 
-	permissions := acls.READ_RESULTS
-	perm, err := acls.CheckAccess(org_config_obj, user_record.Name, permissions)
+	perm, err := acls.CheckAccess(org_config_obj, user_record.Name, acls.SERVER_ADMIN)
 	if !perm || err != nil {
 		return nil, status.Error(codes.PermissionDenied,
 			"User is not allowed to enumerate users.")
 	}
 
+	filterOrgs := true
+
+	if isMemberOfRootOrg(user_record.Orgs) {
```

Comment:
There is an assumption here that a member of the root org will have admin access to all the other orgs which is not currently the case. It is maybe a reasonable assumption in practice though.

I think this function should replicate (or even refactor) the VQL equivalent here 
https://github.com/Velocidex/velociraptor/blob/master/vql/server/users/users.go#L77

Generally we should have once source of truth in the code and call it both from the VQL and API if we are extending the API.

Reply:
I made an assumption early on that the root org was intended to be hierarchically higher than the other orgs, and now I see that assumption is invalid.

Reply:
Kind of - for convenience an admin of the root org automatically gets the org admin permission as well
https://github.com/Velocidex/velociraptor/blob/master/acls/roles.go#L101

so we dont need a special role just for org admins but it is always possible to give users individual permissions and that permission is what actually matters